src/AdminGuesser.js

```diff
@@ -71,11 +91,76 @@ AdminGuesser.defaultProps = {
       },
     },
   }),
-  appLayout: Layout,
+
+  resourceSchemaAnalyzer,
+
+  children,
+  ...rest
+}) => {
+  const [resources, setResources] = useState();
+  const [fetching, setFetching] = useState(true);
+  const [error, setError] = useState();
+
+  if (appLayout) {
+    console.warn(
+      'You are using deprecated prop "appLayout", it was replaced by "layout", see https://github.com/marmelab/react-admin/issues/2918',
+    );
+  }
+  if (loginPage === true && process.env.NODE_ENV !== 'production') {
+    console.warn(
+      'You passed true to the loginPage prop. You must either pass false to disable it or a component class to customize it',
+    );
+  }
+  if (locale) {
+    console.warn(
+      'You are using deprecated prop "locale". You must now pass the initial locale to your i18nProvider',
+    );
+  }
+
+  useEffect(() => {
+    dataProvider
```

Comment:
I suggest you add a detection for the `dataProvider.introspect()` method existence here, and throw a nice Exception in case a developer uses a non-hydra dataProvider. As this method is not in the standard `dataProvider` signature, you need to help developers a bit.